packages/core/core/src/public/MutableBundleGraph.js

```diff
@@ -119,7 +119,9 @@ export default class MutableBundleGraph extends BundleGraph
 
   removeBundleGroup(bundleGroup: BundleGroup): void {
     for (let bundle of this.getBundlesInBundleGroup(bundleGroup)) {
```

Comment:
I believe this is is necessary (we can't just rely on `removeNode` cleanup from bundle groups to remove bundles) as we now have `reference` edges between bundles. Does that sound right @devongovett?